### DISCUSSION TITLE
TBS Agent Lite LUA script not working on Eachine TX16S EdgeTX 2.7.1

### DISCUSSION THREAD

**Discussion by zedthirteen** [EdgeTX/edgetx#2473] [Q&A]
I have installed a TBS Crossfire Micro TX V2 module into my module bay and connected a Nano RX receiver.

When I run TBS agent lite it shows the splash screen 0.96 and then does not show devices (it was sometimes showing the "XF WiFi" entry) and remains "Looking for devices".

It has never detected the "XF Micro TX" or "XF Nano RX" modules - even after several minutes.  I've set the baud rate to the maximum as described in the TBS documentation but have also tried lower baud rates.

The phone app shows "XF WiFi", "XF Micro TX" and "XF Nano RX"

The status lite on TX module and Nano RX are both solid green.

As yet, with this issue, I have not had the confidence to put this on an airframe.

Has anyone else had this issue and/or can offer help?  I saw an issue reporting very slow response but mine never completes

**Comment by zedthirteen** [comment]
I've been in touch with Mike Chin at TBS and he tells me that CRSF does not work with yaapu telemetry at the moment.

This is probably a problem introduced between EdgeTX 2.6 and 2.7.1 - See: https://github.com/EdgeTX/edgetx/issues/1917

I found that if I created a new model without yaapu then TBS Agent Lite run unhindered and able to find devices

**Reply by pfeerick** [reply]
Yes, there is an issue whereby it seems widget scripts and standalone scripts are both trying to talk to the module/access telemetry at the same time, and one seems to prevent the other from working. However, I don't know where the "CRSF does not work with yaapu telemetry at the moment" comes from - as once you have enabled CRSF support in yappu, it still should work. I did test that it worked with ETX and CRSF at some point (using ELRS as the link IIRC), but I don't have anything running ardupilot so can't test the link again now using Tracer... but I would be somewhat surprised if it didn't still work, just not the two together.

**Reply by zedthirteen** [reply]
Thanks Peter, I wasn't aware I had to enable CRSF in yaapu. Are you able to point to how/where I do this? I can't find it through the widget config screens. Do I need to edit the LUA script?

**Reply by pfeerick** [reply]
There is some information here => https://github.com/yaapu/FrskyTelemetryScript/wiki/Passthrough-over-CRSF-and-ExpressLRS

But it should be something like this
https://user-images.githubusercontent.com/5500713/194699892-32062a88-45f7-495d-a724-b45af25fa214.mp4

**Reply by zedthirteen** [reply]
Thanks, I've done that now but it made no difference when trying to run TBS Agent Lite.  With yaapu running with selected model the TBS Agent Lite fails to find any devices.  When I select a model that doesn't have yaapu it does find devices

**Reply by pfeerick** [reply]
Er... well, naturally... changing a setting in yappu config isn't going to suddenly fix an unrelated problem. But it should have made it so yaapu was working with CRSF telemetry. The workaround atm is to not have yaapu running when using TBS Agent Lite.

**Comment by JimB40** [comment]
Have you updated TBS Crossfire Micro TX V2 module's Wifi firmware to 2.XX version? as described in #1917
There are two separate issues that can cause CRSF devices not appear in Agent
1. Module Wifi firmware < 2.00
2. Yappu widget running in background and 'stealing' data from module.

**Reply by zedthirteen** [reply]
yes, I'm on 2.0.4
